store/postgres/src/deployment_store.rs

```diff
@@ -1577,10 +1607,62 @@ impl DeploymentStore {
                 Ok(())
             })?;
         }
+
+        // check if all indexes are valid and recreate them if they aren't
+        let mut conn = self.get_conn()?;
+
+        if ENV_VARS.postpone_attribute_index_creation {
+            let namespace = site.namespace.as_str();
+            let index_list = self.load_indexes(site.clone())?;
+            for table in dst.tables.values() {
+                for (ind_name, create_query) in index_list.indexes_for_table(
+                    &namespace.to_string(),
+                    &table.name.to_string(),
+                    table,
+                    true,
+                    true,
+                ) {
+                    if let Some(index_name) = ind_name {
+                        let table_name = table.name.clone();
+                        let query = r#"
+                            SELECT  x.indisvalid           AS isvalid,
+                                    x.indisready           AS isready
+                            FROM pg_index x
+                                    JOIN pg_class c ON c.oid = x.indrelid
+                                    JOIN pg_class i ON i.oid = x.indexrelid
+                                    LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
+                            WHERE (c.relkind = ANY (ARRAY ['r'::"char", 'm'::"char", 'p'::"char"]))
+                            AND (i.relkind = ANY (ARRAY ['i'::"char", 'I'::"char"]))
+                            AND (n.nspname = $1)
+                            AND (c.relname = $2)
+                            AND (i.relname = $3);"#;
+                        let ii_vec = sql_query(query)
+                            .bind::<Text, _>(namespace)
+                            .bind::<Text, _>(table_name)
+                            .bind::<Text, _>(index_name.clone())
+                            .get_results::<IndexInfo>(&mut conn)?
+                            .into_iter()
+                            .map(|ii| ii.into())
+                            .collect::<Vec<IndexInfo>>();
+                        assert!(ii_vec.len() <= 1);
+                        // Check if the index is valid. If either isvalid or isready flag of pg_index table
+                        // isn't true, drop it and recreate it.
+                        if ii_vec.len() == 0 || !ii_vec[0].isvalid || !ii_vec[0].isready {
+                            if ii_vec.len() > 0 {
+                                let drop_query =
+                                    sql_query(format!("DROP INDEX {}.{};", namespace, index_name));
+                                conn.transaction(|conn| drop_query.execute(conn))?;
+                            }
+                            sql_query(create_query).execute(&mut conn)?;
+                        }
+                    }
+                }
+            }
```

Comment:
This whole `for` loop could also be a method, maybe `IndexList::recreate_invalid_indexes(&self, conn: &mut PgConnection, layout: &Layout)`